### PR TITLE
Remove Send-CommandToVM and fix Mount-Disk function to run as sudo 

### PR DESCRIPTION
**Pull request by asofron**
Affected Tests: 
FCOPY-DISABLE-ENABLE
FCOPY-REPEATED-DELETE
FCOPY-LARGE-FILE 
Test results:

 VHD Under Test        : ubuntu_18.04.1.vhdx
 Total Test Cases      : 3 (3 Pass, 0 Fail, 0 Abort)
 Total Time (dd:hh:mm) : 0:0:23
 XML File              : TestConfiguration
 
    ID TestCaseName                                                 TestResult TestDuration(in minutes) 
 ------------------------------------------------------------------------------------------------------
     1 FCOPY-LARGE-FILE                                                   PASS                 9.45 
     2 FCOPY-REPEATED-DELETE                                              PASS                 5.14 
     3 FCOPY-DISABLE-ENABLE                                               PASS                 9.08

### CODE REVIEW

File: Libraries/CommonFunctions.psm1 [hunk 1]
```diff
@@ -2504,36 +2454,26 @@ function Mount-Disk{
...
-	if (-not $sts) {
-		Write-LogErr "Failed to make file system in the VM $vmName."
-		return $False
-	}
-
-	$sts = Send-CommandToVM -vmPassword $vmPassword -vmPort $vmPort -ipv4 $ipv4  "mount ${driveName}1 /mnt"
-	if (-not $sts) {
-		Write-LogErr "Failed to mount the disk in the VM $vmName."
-		return $False
+$cmdToVM = @"
```
Comment (LiliDeng) [inline Libraries/CommonFunctions.psm1]: indent

File: Libraries/CommonFunctions.psm1 [hunk 2]
```diff
@@ -2504,36 +2454,26 @@ function Mount-Disk{
...
+#!/bin/bash
+# /dev/sdc is used as /dev/sdb is the resource disk by default
+(echo d;echo;echo w)|fdisk /dev/sdc
+(echo n;echo p;echo 1;echo;echo;echo w)|fdisk /dev/sdc
+mkfs.ext4 /dev/sdc1
+mount /dev/sdc1 /mnt
+exit 0
+"@
+$filename = "MountDisk.sh"
```
Comment (LiliDeng) [inline Libraries/CommonFunctions.psm1]: ditto

File: Libraries/CommonFunctions.psm1 [hunk 3]
```diff
@@ -2504,36 +2454,26 @@ function Mount-Disk{
...
+	if (Test-Path ".\${filename}") {Remove-Item ".\${filename}"}
```
Comment (LiliDeng) [inline Libraries/CommonFunctions.psm1]: format
if (Test-Path ".\${filename}") {
    Remove-Item ".\${filename}"
}

File: Libraries/CommonFunctions.psm1 [hunk 4]
```diff
@@ -2504,36 +2454,26 @@ function Mount-Disk{
...
+		"chmod u+x ${filename} && ./${filename}"  -runAsSudo
+	if ($MountDisk){
+		Write-LogInfo "Mounted /dev/sdc1 to /mnt"
+		return $True
 	}
-
-	Write-LogInfo "$driveName has been mounted to /mnt in the VM $vmName."
-	return $True
 }
-
```
Comment (LiliDeng) [inline Libraries/CommonFunctions.psm1]: blank line

File: Libraries/CommonFunctions.psm1 (hunk 1 above)
Comment (asofron) [reply Libraries/CommonFunctions.psm1]: Fixed

File: Libraries/CommonFunctions.psm1 (hunk 3 above)
Comment (asofron) [reply Libraries/CommonFunctions.psm1]: Fixed

File: Libraries/CommonFunctions.psm1 (hunk 2 above)
Comment (asofron) [reply Libraries/CommonFunctions.psm1]: fixed

File: Libraries/CommonFunctions.psm1 (hunk 4 above)
Comment (asofron) [reply Libraries/CommonFunctions.psm1]: fixed

**Comment by LiliDeng** [comment]
Could you please paste test results here?

File: Libraries/CommonFunctions.psm1 [hunk 5]
```diff
@@ -2504,36 +2454,28 @@ function Mount-Disk{
...
-	if (-not $sts) {
-		Write-LogErr "Failed to mount the disk in the VM $vmName."
-		return $False
+	$cmdToVM = @"
+	#!/bin/bash
+	# /dev/sdc is used as /dev/sdb is the resource disk by default
+	(echo d;echo;echo w)|fdisk /dev/sdc
+	(echo n;echo p;echo 1;echo;echo;echo w)|fdisk /dev/sdc
+	mkfs.ext4 /dev/sdc1
+	mount /dev/sdc1 /mnt
```
Comment (chvalean) [inline Libraries/CommonFunctions.psm1]: /mnt is the ephemeral disk, maybe use a subfolder?
And the exit 0 will make the 'if ($MountDisk)' check always work, even if it fails to format/mount?

File: Libraries/CommonFunctions.psm1 [hunk 6]
```diff
@@ -2504,36 +2454,28 @@ function Mount-Disk{
...
+	exit 0
+"@
+	$filename = "MountDisk.sh"
+	if (Test-Path ".\${filename}") {
+		Remove-Item ".\${filename}"
+	}
+		Add-Content $filename "$cmdToVM"
+		Copy-RemoteFiles -uploadTo $ipv4 -port $vmPort -files $filename -username $vmUsername -password $vmPassword -upload
+		$MountDisk = Run-LinuxCmd -username $vmUsername -password $vmPassword -ip $ipv4 -port $vmPort -command  `
+		"chmod u+x ${filename} && ./${filename}"  -runAsSudo
```
Comment (chvalean) [inline Libraries/CommonFunctions.psm1]: double spaces before -runAsSudo

File: Libraries/CommonFunctions.psm1 (hunk 5 above)
Comment (asofron) [reply Libraries/CommonFunctions.psm1]: added if function to exit 1

File: Libraries/CommonFunctions.psm1 (hunk 6 above)
Comment (asofron) [reply Libraries/CommonFunctions.psm1]: Fixed

File: Libraries/CommonFunctions.psm1 (hunk 5 above)
Comment (asofron) [reply Libraries/CommonFunctions.psm1]: added folder /mnt/test

File: Libraries/CommonFunctions.psm1 [hunk 7]
```diff
@@ -2504,36 +2454,36 @@ function Mount-Disk{
...
-	$sts = Send-CommandToVM -vmPassword $vmPassword -vmPort $vmPort -ipv4 $ipv4  "mount ${driveName}1 /mnt"
-	if (-not $sts) {
-		Write-LogErr "Failed to mount the disk in the VM $vmName."
-		return $False
+	$cmdToVM = @"
+	#!/bin/bash
+	# /dev/sdc is used as /dev/sdb is the resource disk by default
+	(echo d;echo;echo w)|fdisk /dev/sdc
+	(echo n;echo p;echo 1;echo;echo;echo w)|fdisk /dev/sdc
+	if [ $? -ne 0 ];then
```
Comment (santoshx) [inline Libraries/CommonFunctions.psm1]: You may want use function check_exit_status from utils.sh.

Comment (asofron) [reply Libraries/CommonFunctions.psm1]: no need to use utils.sh

File: Libraries/CommonFunctions.psm1 [hunk 8]
```diff
@@ -2504,36 +2454,36 @@ function Mount-Disk{
...
-		Write-LogErr "Failed to make file system in the VM $vmName."
-		return $False
-	}
-
-	$sts = Send-CommandToVM -vmPassword $vmPassword -vmPort $vmPort -ipv4 $ipv4  "mount ${driveName}1 /mnt"
-	if (-not $sts) {
-		Write-LogErr "Failed to mount the disk in the VM $vmName."
-		return $False
+	$cmdToVM = @"
+	#!/bin/bash
```
[the remark this answers is not in the document]
Comment (asofron) [reply Libraries/CommonFunctions.psm1]: script works fine

**Pull request merged**